docs/zkapps/o1js/gadgets.mdx

```diff
@@ -0,0 +1,100 @@
+---
+title: Gadgets
+hide_title: true
+description: How gadgets work in o1js.
+keywords:
+  - smart contracts
+  - zkapp
+  - gadgets
+  - bitwise
+  - o1js
+  - blockchain
+---
+
+:::info
+
+Please note that zkApp programmability is not yet available on Mina Mainnet, but zkApps can now be deployed to Berkeley Testnet.
+
+:::
+
+# Gadgets
+
+Gadgets are amalgamations of the lowest level gates in the proof system (addition, multiplication) into more sophisticated boolean operations (XOR, NOT). Gadgets in o1js simplify the process of creating new cryptographic primitives and make it possible to implement other crypto algorithms, such as SHA. 
+
+<!--- I'm grabbing context from other sources SWAG, please review carefully--->
+Think of gadgets as atomic logic gates that mask large circuits. 
+
+Gadgets are small, reusable building blocks you can use to create more complex cryptographic constructions. 
+
+Leverage the properties and behavior of these smaller components to build secure and efficient systems.
+
+Questions: 
+- are all gadgets bitwise provers? do they all use ZkProgram? what does that mean? https://github.com/o1-labs/o1js/blob/main/src/examples/zkprogram/gadgets.ts#L18 
+- are all gadgets used/called as a method that moves the result into one of our 8 fields?
+- handle the constant case by using the added functionality in the bindings. For the prover case, how to describe? or go into detail for each gadget?
+- do we want to list gadgets in the gadgets namespace separately from gadgets not in the gadgets namespace? (where are these gadgets?)
+- do we have examples other than https://github.com/o1-labs/o1js/blob/main/src/examples/zkprogram/gadgets.ts? 
+- include code snippets? 
```

Comment:
I'm happy to sit with you to put together some snippets prior to submitting a PR for review, @barriebyron. Please collate examples from the links Gar supplied as a starting point and let me know when you're ready to review